cabby/geo/walk.py

```diff
@@ -98,10 +98,27 @@ def get_start_poi(map: map_structure.Map, end_point: GeoDataFrame) -> \
     dist = map.poi['centroid'].apply(
         lambda x: util.get_distance_km(end_point['centroid'], x))
 
-    # Get POI within a distance range.
-    # TODO (https://github.com/googleinterns/cabby/issues/25): Change to path
-    # distance.
-    within_distance = map.poi[(dist > 0.2) & (dist < 0.8)]
+
+    # Get closest nodes to points.
+    dest = ox.get_nearest_node(
+        map.nx_graph, util.tuple_from_point(end_point['centroid']))
+
+    # Find nodes whithin 2000 meter path distance.
+    outer_circle_graph = ox.truncate.truncate_graph_dist(
+        map.nx_graph, dest, max_dist=2000, weight='length')
+
+    # Get graph that is too close (less than 200 meter path distance)
+    inner_circle_graph = ox.truncate.truncate_graph_dist(
+        map.nx_graph, dest, max_dist=200, weight='length')
+
+    outer_circle_graph_osmid = [x for x in outer_circle_graph.nodes.keys()]
+    inner_circle_graph_osmid = [x for x in inner_circle_graph.nodes.keys()]
+
+    osmid_in_range = [
+        osmid for osmid in outer_circle_graph_osmid if osmid not in inner_circle_graph_osmid]
+
+    within_distance = map.poi[map.poi['node'].isin(osmid_in_range)]
+
```

Comment:
I think of this more as a 'ring' or 'donut'.  Might be useful to cast variable names in terms of one of those terms?